src/noflo-component-access.js

```diff
@@ -0,0 +1,138 @@
+// noflo-component-access.js
+
+var _ = require('underscore');
+var noflo = require('noflo');
+
+module.exports = facadeComponent;
+
+/**
+ * Create a noflo.Component facade from a given component. Access to the component
+ * is limited to only select properties and functions.
+ * Usage:
+ *  var component = componentAccess(new noflo.Component());
+ */
+function facadeComponent(component) {
+    return _.extend({
+        get inPorts() {
+            return _.mapObject(_.pick(component.inPorts, isInPort), facadePort);
+        },
+        get outPorts() {
+            return _.mapObject(_.pick(component.outPorts, isOutPort), facadePort);
+        },
+        get rpf() {
+            return component.rpf = component.rpf || {};
+        }
+    }, facadeFunctions(component,
+        'getDescription',
+        'getIcon'
+    ), facadeEventEmitter(component));
+}
+
+/**
+ * Create a noflo Port facade from a given noflo Port that includes only a limited
+ * subset of properties and functions.
+ */
+function facadePort(port) {
+    return _.extend({
+        // port.nodeInstance might not be assigned yet
+        get nodeInstance() {
+            return port.nodeInstance && facadeComponent(port.nodeInstance)
+        },
+        get name() {
+            return port.name;
+        }
+    }, facadeFunctions(port,
```

Comment:
Might be good to add a comment with a link to the noflo documentation that describes these attributes.